plugins/package-managers/node/src/main/kotlin/PackageJson.kt

```diff
@@ -141,11 +145,21 @@ data class PackageJson(
 
 private object AuthorListSerializer : JsonTransformingSerializer<List<Author>>(serializer<List<Author>>()) {
     override fun transformDeserialize(element: JsonElement): JsonElement =
-        // TODO: The string from the JSON primitive could be parsed into the dedicated author properties.
         when (element) {
             is JsonObject -> JsonArray(listOf(element))
-            is JsonPrimitive -> JsonArray(listOf(element.wrapPrimitiveInObject("name")))
-            is JsonArray -> JsonArray(element.map { it.wrapPrimitiveInObject("name") })
+            is JsonPrimitive -> JsonArray(listOf(element.toAuthorObject()))
+            is JsonArray -> JsonArray(element.map { it.toAuthorObject() })
+        }
+
+    private fun JsonElement.toAuthorObject(): JsonElement =
```

Comment:
Would it make sense to align the name, e.g. `wrapPrimitiveAuthorInObect()`?

Reply:
I chose this name because strict speaking it's not *always* wrapping a primitive, e.g. if the element already is an object, and because I'm inlining the `wrapPrimitiveInObject` function again in the last commit.